### ISSUE TITLE
improve twisted.words.im.basechat docstrings

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2458]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2458|
|Type|enhancement|
|Created|2007-02-18 21:32:27Z|
|Branch|https://github.com/twisted/twisted/tree/basechat-docstrings-2458-2|

They're sparse and missing in some places.

<details><summary>Searchable metadata</summary>

```
trac-id__2458 2458
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_basechat_docstrings_2458_2 branches/basechat-docstrings-2458-2
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__words words
keywords__documentation documentation
time__1171834347000000 1171834347000000
changetime__1322452237000000 1322452237000000
version__None None
owner__jesstess jesstess
cc__thijs cc__jesstess cc__binjured
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_2"></a>|@jesstess commented|
|-|-|

(In [28803]) Branching to 'basechat-docstrings-2458'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess removed owner|
|-|-|

The branch:
 * Puts docstring quotes on lines by themselves
 * removes debugging print statements
 * remove commented code
 * use standard-compliant spacing between functions and between classes.
 * adds missing epytext markup to all the classes and functions

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @jesstess|
|-|-|

Thanks!

  1. I think that `ContactsList.contacts` and `ContactsList.onlineContacts` probably have `IPerson` _provider_ values.
  1. A few places give the type of a variable as an interface.  This is incomplete; what is probably meant is that the variable must _provide_ the interface, so it should say that.  eg, instead of _@type client: L{Client<interfaces.IClient>}_ it should say _@type client: L{Client<interfaces.IClient>} provider_.
  1. `GroupConversation.group` is documented a couple times as an _implementer_, but probably _provider_ is correct.
  1. Likewise for `ChatUI..persons`, `ChatUI.groups`, and `ChatUI.onlineClients`.  Instead of _are interface_, say _provide interface_.
  1. The docstring for `ChatUI.__init__` looks mostly redundant with the information in the docstring for `ChatUI`.  Also, it's documenting parameters that `ChatUI.__init__` doesn't even accept.
  1. It looks like `ChatUI.getConversation` ends with some invalid epytext, _@returns The conversation window._.  Though, for some reason, this didn't cause the doc builder to go red.  Erf.  Same issue in `getGroupConversation`.
  1. I think we mildly prefer `@return:` over `@returns:`, too.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess removed owner|
|-|-|

Thanks for the review, exarkun. Points 1-7 above should now be addressed in the branch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @jesstess|
|-|-|

Looks good.  The news fragment can be a `.doc` now and the text should probably follow the [documented convention](http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles), otherwise I have no changes to suggest.  The branch does appear to have been created at a point when some conch tests were failing.  It'd probably be good to merge forward and run the new branch on buildbot, just to make sure everything's working right.  Assuming that goes well, please merge.

**Comment by twisted-trac** [comment]
|<img alt="binjured's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|binjured commented|
|-|-|

 Pinging this to see if jesstess intends to make these final minor changes or if it should be unassigned. Thanks for all the work thus far!

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess commented|
|-|-|

(In [33224]) Branching to 'basechat-docstrings-2458-2'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra commented|
|-|-|

Use C{dict} instead of L{dict}.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_12"></a>|@jesstess commented|
|-|-|

Merged forward and fixed some pydoctor warnings.

Build results [here](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/basechat-docstrings-2458-2).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set owner to @jesstess|
|-|-|

Thanks.

  1. I don't like the `L{Person<interfaces.IPerson>}` convention established in this module.  Let's just go with the unsurprising `L{IPerson<interfaces.IPerson>}`.
  1. `ContactsList`'s class docstring uses `param` where it should use `ivar`
  1. `Conversation.sendText` and `GroupConversation.sendText` appear to lie about their return types.  They each return `None` but claim to return a `Deferred`.
  1. `GroupConversation.setTopic` uses `@param:` where it means `@param topic`
  1. in `ChatUI.getConversation`, `Class` is documented as an _IConversation provider class_.  A class the instances of which provide an interface is said to _implement_ that interface, so please change this to _IConversation implementor_.
  1. Likewise for `getGroupConversation`

Thanks!  Please merge when you're happy with the resolution to these points.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_14"></a>|@jesstess commented|
|-|-|

(In [33259]) Address review comments by exarkun.

refs [#2458](https://github.com/twisted/twisted/issues/2458).

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_15"></a>|@jesstess set status to `closed`|
|-|-|

(In [33260]) Merge basechat-docstrings-2458-2

Author: jesstess
Reviewer: exarkun
Fixes: [#2458](https://github.com/twisted/twisted/issues/2458)

Improve the twisted.words.im.basechat API documentation.
